codingTest_jye/신규-아이디-추천.js

```diff
@@ -0,0 +1,41 @@
+// 시뮬레이션 유형
+// 입력값 문자열 - 출력값 문자열
+// 정규표현식을 이용한 풀이를 해야한다.
+function solution(new_id) {
+    //우선 사옹자가 조건을 지켰는지 체크 지켰으면 그냥 통과시키고 안지켰으면 내가 아이디를 하나 정해서 반환해줘야함.
+    //3~15 글자수 제한 확인 AND 정해진 문자만 사용했는지 확인 AND .이 양쪽끝에 있는지 없는지 확인 AND .이 2개이상 연속하는지 확인
+    if (new_id.match(/^.{3,15}$/) && !(/[^a-z0-9-_.]/.test(new_id)) && !(/^\.|\.$/.test(new_id)) && !(/\.{2,}/).test(new_id)) {
+        return new_id;
+    }
+    //1. 문자열의 소문자화.
+    new_id = new_id.toLowerCase();
+    //2. 알파벳 소문자, 숫자, 뺴기, 밑줄, 마침표 제외하고 모든 문자 제거
+    //문자열.match(/정규식/플래그) 문자열에서 정규식에 매칭되는 항목들을 배열로 반환
+    new_id = new_id.match(/[a-z0-9-_.]/gm); //얘가 여기서 배열로 반환하니깐
+    new_id = new_id.join(''); //여기서 다시 문자열로 만들어주고
```

Comment:
replace 메서드 쓰시면 문자열로 리턴되기 때문에 다시 문자열로 안 만들어도 됩니다.
```suggestion javascript
    new_id = new_id.replace(/[^a-z0-9-_.]/g, '');
```

Reply:
repalce 메서드 확인했습니다. 알겠습니다!!